### DISCUSSION TITLE
nginx-proxy with basic auth for portainer container

### DISCUSSION THREAD

**Discussion by rucksman** [nginx-proxy/nginx-proxy#2559] [Q&A]
I am running nginx-proxy successfully for different vhosts in different containers. Also the portainer container does work well behind nginx-proxy.

No I wanted to secure portainer by adding a basic auth to it. So I followed the instructions here https://hub.docker.com/r/jwilder/nginx-proxy#basic-authentication-support and created the file in the correct place. At first it looked good as the basic auth windows appeared when opening portainer in the browser. I entered the credentials and the portainer login page appeared. I entered the portainer credentials, and then the problem occured: The basic auth window appeared again in an endless loop after entering the basic auth credentials over and over again and I was stuck on the portainer login page. After removing the htpasswd file everything worked fine again.

Any help how to get nginx-proxy and the portainer container to work with basic auth?

My portainer.mydomain.tld_location looks like this
```
proxy_set_header Connection $http_connection;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "upgrade";
```

**Comment by OskarBorek** [comment]
i have the same problem...any ideas?

**Comment by nicosapp** [comment]
same here!
